### PR TITLE
VCB-186 Flag instructions

### PR DESCRIPTION
**Pull request by taggatips**

### CODE REVIEW

File: src/instruction/instructions/flags/msr.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  setBits
+} from 'instruction/opcode'
+import { Halfword, Word } from 'types/binary'
+import { BaseInstruction } from '../base'
+
+export class MsrInstruction extends BaseInstruction {
+  public name: string = 'MSR'
+  public pattern: string = '111100111000XXXX'
+  private patternSecondHalf: string = '1000100000000000'
+  private apsrPattern = 'XXXX000000000000'
```
Comment (ni507) [inline src/instruction/instructions/flags/msr.ts:17]: This variable is not needed anymore.

**Review by ni507 — CHANGES_REQUESTED** [review]
Everything else looks good and clean to me.

**Comment by m-dot-wigger** [comment]
image
is s not relevant for us within pattern. 
If then pattern should be updated with X there so during execution it does not matter if there are ones or zeros there to find the Instruction to Execute the opcode.

File: src/instruction/instructions/flags/msr.ts (hunk 1 above)
Comment (taggatips) [reply src/instruction/instructions/flags/msr.ts:17]: Deleted.

**Pull request merged**